src/js/elements/mapchip.js

```diff
@@ -38,18 +46,48 @@ export class MapChip extends Observable {
     handleChoroplethFilter(args) {
         let primaryGroup = args.data.metadata.primary_group;
         let groups = getFilterGroups(args.data.metadata.groups, primaryGroup);
+        this.groups = groups;
+
+        $(mapOptionsClass).find(`${filterRowClass}[data-isextra=true]`).remove();
+        const filterArea = $(mapOptionsClass).find(filterContentClass);
+        let defaultFilters = [];
+        if (typeof args.filter !== 'undefined') {
```

Comment:
under what circumstances would this be undefined? If you are guarding against incomplete data being sent from the server, then you should create .filter in dataobjects.js so that your code doesn't need to repeatedly check for missing values